In [1]:
import json
import pickle
def load_data(file_path):
    """ Load the data from a pickle file. """
    with open(file_path, 'rb') as f:
        return pickle.load(f)

# KL Only

In [2]:
# 4 label 4 mods
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 38, 178,  21,   0],
       [ 34, 240,  70,   4],
       [  3,  67,  58,   4],
       [  0,  24,  32,   3]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.51      0.16      0.24       237\n > 3 & ≤ 7 days       0.47      0.69      0.56       348\n> 7 & ≤ 14 days       0.32      0.44      0.37       132\n      > 14 days       0.27      0.05      0.09        59\n\n       accuracy                           0.44       776\n      macro avg       0.39      0.34      0.32       776\n   weighted avg       0.44      0.44      0.40       776\n'}, 'f1': {'test': 0.3150011017373762}}}


In [5]:
# TS CXR ECG
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 23, 207,   7,   0],
          [ 16, 325,   7,   0],
          [  6, 121,   5,   0],
          [  1,  52,   6,   0]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.50      0.10      0.16       237\n > 3 & ≤ 7 days       0.46      0.93      0.62       348\n> 7 & ≤ 14 days       0.20      0.04      0.06       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.45       776\n      macro avg       0.29      0.27      0.21       776\n   weighted avg       0.39      0.45      0.34       776\n'},
  'f1': {'test': 0.21088059693612876}}}

In [7]:
## 4 label TS_ECG_Text -- with smooth init=0.5 decay=0.05
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg/TS_ECG_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 28, 190,  19,   0],
       [ 24, 252,  69,   3],
       [  7,  65,  50,  10],
       [  0,  24,  31,   4]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.12      0.19       237\n > 3 & ≤ 7 days       0.47      0.72      0.57       348\n> 7 & ≤ 14 days       0.30      0.38      0.33       132\n      > 14 days       0.24      0.07      0.11        59\n\n       accuracy                           0.43       776\n      macro avg       0.37      0.32      0.30       776\n   weighted avg       0.43      0.43      0.38       776\n'}, 'f1': {'test': 0.3000142750739081}}}


In [6]:
## 4 label TS_CXR_Text -- with smooth init=0.5 decay=0.05
print(load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg/TS_CXR_Text/TS_48/Atten/f1/result.pkl"))

{'42': {'confusion_matrix': {'test': array([[ 39, 190,   8,   0],
       [ 28, 284,  35,   1],
       [  7,  81,  42,   2],
       [  0,  41,  16,   2]])}, 'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.53      0.16      0.25       237\n > 3 & ≤ 7 days       0.48      0.82      0.60       348\n> 7 & ≤ 14 days       0.42      0.32      0.36       132\n      > 14 days       0.40      0.03      0.06        59\n\n       accuracy                           0.47       776\n      macro avg       0.45      0.33      0.32       776\n   weighted avg       0.48      0.47      0.41       776\n'}, 'f1': {'test': 0.3188784488085913}}}


- MissingInd

4 mod // {'42': {'confusion_matrix': {'test': array([[ 949,  693,   51,   36],
          [ 753, 1099,  189,  260],
          [  82,  325,  160,  278],
          [  21,  113,   87,  168]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.53      0.55      0.54      1729\n > 3 & ≤ 7 days       0.49      0.48      0.49      2301\n> 7 & ≤ 14 days       0.33      0.19      0.24       845\n      > 14 days       0.23      0.43      0.30       389\n\n       accuracy                           0.45      5264\n      macro avg       0.39      0.41      0.39      5264\n   weighted avg       0.46      0.45      0.45      5264\n'},
  'f1': {'test': 0.3898733930886193}}}

In [3]:
# TS CXR ECG
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg-missingInd/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[   0, 1663,   61,    5],
          [   1, 1968,  273,   59],
          [   1,  594,  204,   46],
          [   0,  232,  131,   26]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.00      0.00      0.00      1729\n > 3 & ≤ 7 days       0.44      0.86      0.58      2301\n> 7 & ≤ 14 days       0.30      0.24      0.27       845\n      > 14 days       0.19      0.07      0.10       389\n\n       accuracy                           0.42      5264\n      macro avg       0.23      0.29      0.24      5264\n   weighted avg       0.26      0.42      0.31      5264\n'},
  'f1': {'test': 0.23773831551712107}}}

In [4]:
# TS CXR TEXT
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[1104,  552,   62,   11],
          [1000, 1001,  219,   81],
          [ 177,  357,  242,   69],
          [  48,  153,  142,   46]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.64      0.54      1729\n > 3 & ≤ 7 days       0.49      0.44      0.46      2301\n> 7 & ≤ 14 days       0.36      0.29      0.32       845\n      > 14 days       0.22      0.12      0.15       389\n\n       accuracy                           0.45      5264\n      macro avg       0.39      0.37      0.37      5264\n   weighted avg       0.44      0.45      0.44      5264\n'},
  'f1': {'test': 0.3694390134642215}}}

In [4]:
# TS ECG Text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl/los-48-cxr-notes-ecg-missingInd/TS_ECG_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 131, 1582,   16,    0],
          [ 233, 1891,  160,   17],
          [ 163,  509,  156,   17],
          [ 113,  183,   88,    5]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.20      0.08      0.11      1729\n > 3 & ≤ 7 days       0.45      0.82      0.58      2301\n> 7 & ≤ 14 days       0.37      0.18      0.25       845\n      > 14 days       0.13      0.01      0.02       389\n\n       accuracy                           0.41      5264\n      macro avg       0.29      0.27      0.24      5264\n   weighted avg       0.33      0.41      0.33      5264\n'},
  'f1': {'test': 0.24137641260179304}}}

# KL + MI

In [7]:
# 4mod
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[123,  96,  16,   2],
          [111, 149,  66,  22],
          [ 17,  40,  55,  20],
          [  4,  12,  28,  15]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.48      0.52      0.50       237\n > 3 & ≤ 7 days       0.50      0.43      0.46       348\n> 7 & ≤ 14 days       0.33      0.42      0.37       132\n      > 14 days       0.25      0.25      0.25        59\n\n       accuracy                           0.44       776\n      macro avg       0.39      0.40      0.40       776\n   weighted avg       0.45      0.44      0.44       776\n'},
  'f1': {'test': 0.39665579059548317}}}

In [9]:
# TS CXR TEXT
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 79, 153,   5,   0],
          [ 51, 276,  21,   0],
          [  9, 102,  20,   1],
          [  0,  45,  14,   0]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.57      0.33      0.42       237\n > 3 & ≤ 7 days       0.48      0.79      0.60       348\n> 7 & ≤ 14 days       0.33      0.15      0.21       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.48       776\n      macro avg       0.35      0.32      0.31       776\n   weighted avg       0.45      0.48      0.43       776\n'},
  'f1': {'test': 0.30648717417334437}}}

In [10]:
# TS CXR ECG
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg/TS_CXR_ECG/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 13, 224,   0,   0],
          [ 13, 333,   2,   0],
          [  5, 127,   0,   0],
          [  2,  57,   0,   0]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.39      0.05      0.10       237\n > 3 & ≤ 7 days       0.45      0.96      0.61       348\n> 7 & ≤ 14 days       0.00      0.00      0.00       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.45       776\n      macro avg       0.21      0.25      0.18       776\n   weighted avg       0.32      0.45      0.30       776\n'},
  'f1': {'test': 0.17696663605754515}}}

In [13]:
# TS ECG TEXT
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg/TS_ECG_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[151,  68,  14,   4],
          [147, 139,  45,  17],
          [ 19,  55,  36,  22],
          [  1,  26,  23,   9]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.47      0.64      0.54       237\n > 3 & ≤ 7 days       0.48      0.40      0.44       348\n> 7 & ≤ 14 days       0.31      0.27      0.29       132\n      > 14 days       0.17      0.15      0.16        59\n\n       accuracy                           0.43       776\n      macro avg       0.36      0.37      0.36       776\n   weighted avg       0.43      0.43      0.42       776\n'},
  'f1': {'test': 0.35785330613632504}}}

- MissingInd

In [3]:
# 4 mod
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text_ECG/TS_48/Atten/f1/result.pkl") 

{'42': {'confusion_matrix': {'test': array([[ 973,  649,   67,   40],
          [ 744, 1002,  303,  252],
          [  95,  260,  209,  281],
          [  24,   92,  103,  170]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.53      0.56      0.55      1729\n > 3 & ≤ 7 days       0.50      0.44      0.47      2301\n> 7 & ≤ 14 days       0.31      0.25      0.27       845\n      > 14 days       0.23      0.44      0.30       389\n\n       accuracy                           0.45      5264\n      macro avg       0.39      0.42      0.40      5264\n   weighted avg       0.46      0.45      0.45      5264\n'},
  'f1': {'test': 0.39639216265087196}}}

In [4]:
# ts-cxr-text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg-missingInd/TS_CXR_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[ 416, 1274,   29,   10],
          [ 314, 1709,  221,   57],
          [  34,  535,  213,   63],
          [  13,  204,  128,   44]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.54      0.24      0.33      1729\n > 3 & ≤ 7 days       0.46      0.74      0.57      2301\n> 7 & ≤ 14 days       0.36      0.25      0.30       845\n      > 14 days       0.25      0.11      0.16       389\n\n       accuracy                           0.45      5264\n      macro avg       0.40      0.34      0.34      5264\n   weighted avg       0.45      0.45      0.42      5264\n'},
  'f1': {'test': 0.33811434089606607}}}

In [5]:
# ts-ecg-text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/kl_mi/los-48-cxr-notes-ecg-missingInd/TS_ECG_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[  76, 1595,   40,   18],
          [ 141, 1786,  252,  122],
          [  92,  418,  210,  125],
          [  71,  136,  106,   76]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.20      0.04      0.07      1729\n > 3 & ≤ 7 days       0.45      0.78      0.57      2301\n> 7 & ≤ 14 days       0.35      0.25      0.29       845\n      > 14 days       0.22      0.20      0.21       389\n\n       accuracy                           0.41      5264\n      macro avg       0.31      0.32      0.29      5264\n   weighted avg       0.34      0.41      0.34      5264\n'},
  'f1': {'test': 0.2855378630610964}}}

{'42': {'confusion_matrix': {'test': array([[   0, 1565,  158,    6],
          [   0, 1707,  572,   22],
          [   1,  432,  382,   30],
          [   0,  129,  237,   23]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.00      0.00      0.00      1729\n > 3 & ≤ 7 days       0.45      0.74      0.56      2301\n> 7 & ≤ 14 days       0.28      0.45      0.35       845\n      > 14 days       0.28      0.06      0.10       389\n\n       accuracy                           0.40      5264\n      macro avg       0.25      0.31      0.25      5264\n   weighted avg       0.26      0.40      0.31      5264\n'},
  'f1': {'test': 0.25066617581765976}}}

# No Weights

In [5]:
# TS ECG Text
load_data("/data/wang/junh/results/Fuse_moe/all_los/multiclass/final/noWeights/los-48-cxr-notes-ecg/TS_ECG_Text/TS_48/Atten/f1/result.pkl")

{'42': {'confusion_matrix': {'test': array([[124,  95,  18,   0],
          [116, 156,  76,   0],
          [ 17,  49,  66,   0],
          [  3,  17,  39,   0]])},
  'classification_report': {'test': '                 precision    recall  f1-score   support\n\n       ≤ 3 days       0.48      0.52      0.50       237\n > 3 & ≤ 7 days       0.49      0.45      0.47       348\n> 7 & ≤ 14 days       0.33      0.50      0.40       132\n      > 14 days       0.00      0.00      0.00        59\n\n       accuracy                           0.45       776\n      macro avg       0.33      0.37      0.34       776\n   weighted avg       0.42      0.45      0.43       776\n'},
  'f1': {'test': 0.34173960922475544}}}